### ISSUE TITLE
In inline math mode, allow breaking lines after a comma

### ISSUE THREAD

**Issue by jmigual** [typst/typst#5315]
### Description

There are already some cases where inline math breaks lines automatically. However, it seems that comma doesn't trigger this scenario. It'd be nice to either: (1) have an option to choose whether or not a line break can be added after a comma or (2) make commas "breakable" by default.

### Use Case

In my use case, I was trying to write a very long tuple:

```typst
Test of a sentence with math that should break after a comma: 
$E = (M, O, sans("cap"), sans("prc"), sans("nextM"), sans("csiM"), sans("csd"))$
```

And this is how it rendered:

![image](https://github.com/user-attachments/assets/62faef9d-f6ef-461f-8394-e357eeef6a34)

The Typst version used for this is v0.12.0.

**Comment by Enivex** [comment]
I doubt the problem here is commas. Rather, it's the matching delimiters. In any case I'd strongly recommend considering using display math if you have something this long

**Comment by laurmaedje** [comment]
The math linebreaking rules were defined in https://github.com/typst/typst/pull/1826 and adopted from TeX. From a quick experiment, linebreaking at commas in math is also not allowed in LaTeX, probably (?) for some good reason.

**Comment by jmigual** [comment]
Indeed. However, in LaTeX you can use `\allowbreak` right after a comma and then the engine can consider line breaks after it. Just having that would be enough for my case.

**Comment by YDX-2147483647** [comment]
(Adapted from [How to control line breaks (wrap opportunities) in inline math? - Questions - Typst Forum](https://forum.typst.app/t/how-to-control-line-breaks-wrap-opportunities-in-inline-math/3562))

### A robust minimal example

```typst
#set page(height: auto, width: 10em, margin: 1em)
Then $exists x > 0, f(x) = y$ and why?
```

![](https://forum.typst.app/uploads/default/original/2X/c/cb1c197fd8bb631ea658a2e533a2c150db0e3f23.png)

### Workaround

Split it into two parts:

```typst
$exists x > 0,$ $f(x) = y$
```

Or use [`allowbreak` defined below by N. V. Lang](https://github.com/typst/typst/issues/5315#issuecomment-3094644967).

Related: https://github.com/typst/typst/issues/4795

**Comment by nvlang** [comment]
> Indeed. However, in LaTeX you can use `\allowbreak` right after a comma and then the engine can consider line breaks after it. Just having that would be enough for my case.

I was facing essentially the same issue, and I've found this workaround to be fairly usable in some cases:

```typst
#let allowbreak = math.script(math.class("binary", ""))
```

I came up with this after looking at some of Typst's source code, which hints that:

1.   Binary operators are treated as opportunities for line breaks:

     https://github.com/typst/typst/blob/b790c6d59ceaf7a809cc24b60c1f1509807470e2/crates/typst-layout/src/math/run.rs#L304-L312

2.  Binary operators aren't padded with spaces in script-sized math:

    https://github.com/typst/typst/blob/b790c6d59ceaf7a809cc24b60c1f1509807470e2/crates/typst-layout/src/math/run.rs#L457-L459

Unfortunately, this solution doesn't work well in all cases (for example, it breaks the spacing in `$a allowbreak + b$`). But I personally found it sufficient as a temporary workaround for the commas use case.

(I tried making all commas in math mode automatically insert such an `allowbreak` after themselves with `#show math.comma: math.comma + allowbreak`, but this seemed to break kerning around commas both inside and outside of math mode.)
